content/articles/web-apps.md

```diff
@@ -0,0 +1,122 @@
+
+# Why have a web application?
+..as opposed to making research code available for users to download and run themselves?
+
+Web apps are:
+- **Accessible**: no special skills or coding knowledge are required from the user
+- **Available**: on all devices with a network connection, and have no special requirements of the user's local computer other than a modern browser.
+- **Secure**: password-based access can be provided so only authorised users have access to sensitive data and functionality
+- **Fast**: can be deployed to appropriate compute resources so intensive tasks can be executed as quickly as possible.
+- **Reliable**: can deploy known stable versions of research code - this helps with reproducible research
+- **Easy to use**: we can design the user experience to focus on their concerns e.g. particular practical or policy goals, or a need to explore a research model in a guided way with appropriate visualisations.
+- **Convenient**: can allow users to easily save, manage and share their data and sessions. Users do not need direct access to large datasets required for code execution
+- **A useful starting point for advanced users**: a web app allows other researchers to quickly assess the utility of a model before putting in effort to download, configure and run it locally.
+
+## What about Shiny?
+[Shiny](https://shiny.posit.co) is a popular choice for researchers wishing to provide web access to their models developed in R, but it has some drawbacks.
+
+Pros:
+- rapid web application development within R: easy to integrate with existing R-based research code
+- easy to get started
+- powerful presentation features
+
+Cons:
+- difficult/impossible to automate tests in order to be confident that new features have not broken existing functionality
+- slow
+- does not scale easily
+- limited capabilities for many application features e.g. user authentication, or managing saved user data
+- handles long-running processes poorly, causing the application to hang
+
+
+In our experience, where an application is written over a single period and then not modified going forward, Shiny has proven a good choice.
+Where an application will need to evolve over time as new features are added, the quick startup time of Shiny is replaced by a slow and
+error-prone development process
+
+Shiny's limitations mean that it may not be a good choice for a final application to present to users, but could be a useful prototyping tool,
+allowing the research team to develop ideas for how an application's user interface should be structured and what options it should provide.
+The RESIDE team could then migrate the prototype Shiny app's functionality to a full web application.
+
+However, in cases where a simple application is all that is required, with no complex interactivity or scalability requirements,
+and no need for future changes, a Shiny application may be the best option for the final application too.
```

Comment:
This feels a bit like going in circles. At the start we say shiny is good if the app is not modified going forward and then we move on to web app discussion and then we come back to shiny, maybe just merge the shiny discussion into the first part and web app into the second?